tensorboard/plugins/debugger/tf_debugger_dashboard/tf-op-selector.html

```diff
@@ -480,45 +482,71 @@
       debugWatchChangeMethod(debugWatch, isChecked);
     },
 
-    _handleForceNodeExpansion(forceExpansionNodeName) {
+    // Shared method for handling externally forced expansion of a node, with
+    // optional enabling of the breakpoint by checking the checkbox and optional
+    // highlighting of the node.
+    _handleForceNode(forceNodeName, toCheck, toHighlight) {
       this.set('_filterInput', '');
-
-      if (forceExpansionNodeName == null) {
-        return;
-      }
-      if (this._levelName2Container == null) {
-        return;  // No node tree has been rendered yet.
-      }
-      const levelItems = tf_debugger_dashboard.splitNodeName(forceExpansionNodeName);
-      for (let i = 1; i <= levelItems.length; ++i) {
-        let currLevelName = levelItems.slice(0, i).join('/');
-        const currTreeNode = this._levelName2Node[currLevelName];
-
-        if (i < levelItems.length) {
-          // At non-lowest level.
-          if (this._levelName2Container[currLevelName] != null) {
-            // It is possible that the node is not present in the tree, e.g.,
-            // due to node-name filtering or other types of filtering.
-            this._levelName2Container[currLevelName].setAttribute('opened', true);
-          }
-        } else {
-          // At lowest level.
-          // If the lowest level is a meta node, send enable-breakpoint request
-          // for all its children.
-          if (!currTreeNode.debugWatch) {
-            this._handleMetaNodeChange(currTreeNode, currTreeNode.debugWatchChange, true);
-          }
-
-          currTreeNode.setToAllCheckedExternally();
-          const debugWatch = currTreeNode.debugWatch;
-          if (debugWatch) {
-            if (this._selectedDebugWatchMapping[debugWatch.node_name] == null) {
-              this._selectedDebugWatchMapping[debugWatch.node_name] = debugWatch;
+      // Use set timeout to give the tree list a chance to update.
+      setTimeout(() => {
+        if (forceNodeName == null) {
+          return;
+        }
+        if (this._levelName2Container == null) {
+          return;  // No node tree has been rendered yet.
+        }
+        const levelItems = tf_debugger_dashboard.splitNodeName(forceNodeName);
+        for (let i = 1; i <= levelItems.length; ++i) {
+          let currLevelName = levelItems.slice(0, i).join('/');
+          const currTreeNode = this._levelName2Node[currLevelName];
+
+          if (i < levelItems.length) {
+            // At non-lowest level.
+            if (this._levelName2Container[currLevelName] != null) {
+              // It is possible that the node is not present in the tree, e.g.,
+              // due to node-name filtering or other types of filtering.
+              this._levelName2Container[currLevelName].setAttribute('opened', true);
+            }
+          } else {
+            // At lowest level.
```

Comment:
It might be worth moving this logic out of the for loop and making the loop's termination condition `< levelItems.length`

Reply:
The two cases (< levelItems.length and == levelItems.length) share some common variables, e.g., currTreeNode. So leaving them in the same for loop seems to save some duplicate code.